src/main/java/co/cask/hydrator/salesforce/plugin/source/streaming/SalesforcePushTopicListener.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2019 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.hydrator.salesforce.plugin.source.streaming;
+
+import co.cask.hydrator.salesforce.SalesforceConstants;
+import co.cask.hydrator.salesforce.authenticator.AuthResponse;
+import co.cask.hydrator.salesforce.authenticator.Authenticator;
+import co.cask.hydrator.salesforce.authenticator.AuthenticatorCredentials;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+import org.cometd.client.BayeuxClient;
+import org.cometd.client.transport.LongPollingTransport;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.util.ssl.SslContextFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Listens to a specific Salesforce pushTopic and adds messages to the blocking queue,
+ * which can be read by a user of the class.
+ */
+public class SalesforcePushTopicListener {
+  private static final Logger LOG = LoggerFactory.getLogger(SalesforcePushTopicListener.class);
+
+  private static final String DEFAULT_PUSH_ENDPOINT = "/cometd/" + SalesforceConstants.API_VERSION;
+  /**
+   * Timeout of 110 seconds is enforced by Salesforce Streaming API and is not configurable.
+   * So we enforce the same on client.
+   */
+  private static final int CONNECTION_TIMEOUT = 110;
+  private static final long HANDSHAKE_TIMEOUT_MS = TimeUnit.SECONDS.toMillis(110);
```

Comment:
Comment says 110 seconds but code says 110 milliseconds. Which one is correct?

Reply:
It's 110 seconds. This line TimeUnit.SECONDS.toMillis(110)
just transforms 110 seconds in 110000 since we need to use milliseconds in code below. Does it make sense?

Reply:
ah yes my bad, read the code too quickly